Sources/MapboxMaps/Camera/SimpleCameraAnimator.swift

```diff
@@ -0,0 +1,214 @@
+                  mapboxMap: MapboxMapProtocol,
+                  cameraOptionsInterpolator: CameraOptionsInterpolatorProtocol,
+                  dateProvider: DateProvider) {
+        self.from = from
+        self.to = to
+        self.duration = duration
+        self.unitBezier = UnitBezier(p1: curve.p1, p2: curve.p2)
+        self.owner = owner
+        self.mapboxMap = mapboxMap
+        self.cameraOptionsInterpolator = cameraOptionsInterpolator
+        self.dateProvider = dateProvider
+        super.init()
+    }
+
+    internal func startAnimation() {
+        startAnimation(afterDelay: 0)
+    }
+
+    /// Starts the animation.
+    ///
+    /// This method sets ``BasicCameraAnimator/state`` to `.active` immediately regardless
+    /// of `delay`. It also call the delegate to indicate that it has started running. Does nothing if `state`
+    /// is not `.inactive`.
+    /// - Parameter delay: An amount of time to wait before beginning to update the map's camera.
+    internal func startAnimation(afterDelay delay: TimeInterval) {
+        switch internalState {
+        case .initial:
+            internalState = .running(startDate: dateProvider.now + delay)
+        case .running:
+            // already running; do nothing
+            break
+        case .final:
+            // animators cannot be restarted
+            break
+        }
+    }
+
+    /// Cancels the animation.
+    ///
+    /// This method sets ``BasicCameraAnimator/state`` to `.inactive`, informs the delegate
+    /// that the animation has stopped running, and clears and invokes
+    /// ``SimpleCameraAnimator/completion``. Does nothing if `state` is not `.active`.
+    internal func cancel() {
+        switch internalState {
+        case .initial, .running:
+            internalState = .final
+            invokeCompletionBlocks(with: .current) // `current` represents an interrupted animation.
+        case .final:
+            // Already stopped, so do nothing
+            break
+        }
+    }
+
+    /// An alias for ``SimpleCameraAnimator/cancel()``.
+    internal func stopAnimation() {
+        cancel()
+    }
+
+    /// Adds completion to the list of completion handlers to be invoked when the animation completes or is
+    /// canceled. If the animation has already completed or been canceled, this method does nothing.
+    /// The animator only holds a strong reference to the handler until it finishes or is canceled.
+    /// - Parameter completion: A handler to invoke when the animator completes.
+    internal func addCompletion(_ completion: @escaping AnimationCompletion) {
+        switch internalState {
+        case .initial, .running:
+            completionHandlers.append(completion)
+        case .final:
+            // already stopped, so do nothing
+            break
+        }
+    }
+
+    private func invokeCompletionBlocks(with position: UIViewAnimatingPosition) {
+        let handlers = completionHandlers
```

Comment:
What is the reason to capture handlers here? Multithreading?

Reply:
Good call. It's unnecessary.